### ISSUE TITLE
aria-invalid="False" is hard-coded to textboxes

### ISSUE THREAD

**Issue by eden-jh** [umbraco/Umbraco-CMS#12427]
### Which *exact* Umbraco version are you using? For example: 9.0.1 - don't just write v9

9.5.0-rc

### Bug summary

The textbox property editor has `aria-invalid="False"` hard-coded into the view. This means that it's not updated when the textbox has an error. Also, because "False" is capitalized, Firefox exposes the input as invalid.

### Specifics

If a mandatory textbox is left blank, or if the textbox exceeds its character limit, the value of `aria-invalid` is never updated to reflect this.

https://github.com/umbraco/Umbraco-CMS/blob/78786e942fa302b011ed9e4782a28d1f32f6f585/src/Umbraco.Web.UI.Client/src/views/propertyeditors/textbox/textbox.html#L3-L11

![A textbox property error with a value exceeding its character limit. It shows a message that says "Maximum 5 characters, 1 too many'. Inspector is open to show that the input element has the aria-invalid attribute set as False, with a capital F](https://user-images.githubusercontent.com/55098579/169104965-da2cb9ab-474a-49d7-8696-acd1c9a799e7.png)

Also, because the "F" in "False" is capitalized, Firefox exposes the form control state as invalid. 

<img src="https://user-images.githubusercontent.com/55098579/169105357-b1b09b12-236c-4272-95da-487054bed3e8.png" alt="Firefox dev tools Accessibility panel, with the properties panel expanded. The selected element has a role of 'entry' and a name of 'Property A (No Validation)'. Its states property is expanded to show that one of its states is 'invalid'" width="400"/>

This is in line with the [ARIA spec](https://w3c.github.io/aria/#aria-invalid), which says that "Any value not recognized in the list of allowed values MUST be treated by user agents as if the value `true` had been provided.".

It's possible this is the case for other browsers too, but the only other one I've checked is Chrome which exposes `aria-invalid="False"` as Invalid user entry: false (at least in the version I'm using).

### Steps to reproduce

1. Create a Page document type. Give it three properties with a textbox property editor
   1. Property A: No validation
   2. Property B: Set maximum characters to 5
   3. Property C: Mandatory
4. Add a Page content node
5. Enter 6 characters for property C. Notice the message says "Maximum 5 characters, 1 too many" but the input still has `aria-invalid="False"`
6. Click Save and publish
7. Notice that Property C shows error messages and uses error styles, but the input still has `aria-invalid="False"`
8. Notice that Property B also still has `aria-invalid="False"`
9. Right-click the input element for Property A and choose Inspect Accessibility Properties from the context menu. (You can also find the input element in the inspector first to confirm you're looking at the right one, then right-click that element in inspector and Inspect Accessibility Properties)
10. In the Accessibility properties, expand the states properties and notice that one of the states is "invalid"
11. If you go back to inspector and change `aria-invalid` to all-lowercase `false`, you'll be able to see that the invalid state is removed

#### Expected result

When a required field error is showing (like step 7 of the steps to reproduce), `aria-invalid` should be `true`. Also, when there are client-side errors like the character limit, `aria-invalid` should be `true`. This change can happen when the error is initially detected (at step 6 above), when the input is blurred after the error is detected, or when content is saved/published. (Whatever is most consistent with other property editors.)

Also, when fields are not invalid, the `false` value of the `aria-invalid` attribute should be all lowercase.

#### Actual result
Textbox inputs always have `aria-invalid="False"` with a capital F in "False", regardless of the error status.

**Comment by bjarnef** [comment]
It seems this hasn't changed since it was added in v8.1: https://github.com/umbraco/Umbraco-CMS/pull/5664
Not sure why it doesn't use the following:

```
aria-invalid="{{textboxFieldForm.textbox.$invalid ? true : false}}
```

It would probably also make sense to add to other property editors like `textarea`:
https://github.com/umbraco/Umbraco-CMS/blob/v10/contrib/src/Umbraco.Web.UI.Client/src/views/propertyeditors/textarea/textarea.html

**Comment by umbrabot** [comment]
Hi @eden-jh,

We're writing to let you know that we would love some help with this issue. We feel that this issue is ideal to flag for a community member to work on it. Once flagged here, folk looking for issues to work on will know to look at yours. Of course, please feel free work on this yourself ;-).  If there are any changes to this status, we'll be sure to let you know. 

For more information about issues and states, have a look at [this blog post](https://umbraco.com/blog/the-umbraco-issue-tracker-process/)

Thanks muchly, from your friendly Umbraco GitHub bot :-)
